esx_service/vmdk_ops_test.py

```diff
@@ -287,6 +289,62 @@ def test_failure(self):
             with self.assertRaises(vmdk_ops.ValidationError):
                 vmdk_ops.validate_opts(opts, self.path)
 
+def create_vm(si, vm_name, datastore):
+        content = si.RetrieveContent()
+        datacenter = content.rootFolder.childEntity[0]
+        vm_folder = datacenter.vmFolder
+        hosts = datacenter.hostFolder.childEntity
+        resource_pool = hosts[0].resourcePool
+
+        datastore_path = '[' + datastore + '] ' + vm_name
+
+        # bare minimum VM shell, no disks. Feel free to edit
+        vmx_file = vim.vm.FileInfo(logDirectory=None,
+                                   snapshotDirectory=None,
+                                   suspendDirectory=None,
+                                   vmPathName=datastore_path)
+
+
+        config = vim.vm.ConfigSpec( 
+                                name=vm_name, 
+                                memoryMB=128, 
+                                numCPUs=1,
+                                files=vmx_file, 
+                                guestId='rhel5_64Guest', 
+                                version='vmx-11'
+                              )
+
+        task = vm_folder.CreateVM_Task(config=config, pool=resource_pool)
+        vmdk_ops.wait_for_tasks(si, [task])
+
+        vm = [d for d in si.content.rootFolder.childEntity[0].vmFolder.childEntity 
```

Comment:
Badically, SearchIndex.FindChild does exactly the same as your code (scans the folder to match the name) but I suspect it does it on the server, so with many VMs I think it would provide a significant difference in perf. 

```python
name = 'photon1'
dc = si.content.rootFolder.childEntity[0]
si.content.searchIndex.FindChild(dc.vmFolder, name)
```